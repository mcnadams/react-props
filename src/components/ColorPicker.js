import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styles from '../index.css';

export default class ColorPicker extends PureComponent {

  static propTypes = { 
    color: PropTypes.string.isRequired
  }

  clickHandler = () => {
    console.log(this.props.color);
  }

  render() {
    return (
      <button onClick={this.clickHandler} className={styles[this.props.color]}>{this.props.color}</button>
    );
  }

}
