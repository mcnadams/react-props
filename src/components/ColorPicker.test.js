import React from 'react';
import { shallow } from 'enzyme';
import ColorPicker from './ColorPicker';

describe('Color Picker', () => {
  it('renders a Color Button', () => {
    const wrapper = shallow(<ColorPicker color="red"/>);
    expect(wrapper).toMatchSnapshot();
  });
});
