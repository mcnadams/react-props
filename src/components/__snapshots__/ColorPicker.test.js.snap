// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color Picker renders a Color Button 1`] = `
<button
  className="red"
  onClick={[Function]}
>
  red
</button>
`;
